assets/data_table/main.js

```diff
@@ -585,32 +560,42 @@ function App({ ctx, data }) {
         renderLayer(
           <HeaderMenu
             layerProps={layerProps}
+            menu={menu}
             orderBy={orderBy}
+            order={content.order}
             selectAllCurrent={selectAllCurrent}
             hasFiltering={hasFiltering}
             hasSorting={hasSorting}
             filterBy={filterBy}
-            currentMenuForm={currentMenuForm}
-            setCurrentMenuForm={setCurrentMenuForm}
+            filters={content.filters}
           />
         )}
       {!hasData && <p className="no-data">No data</p>}
       {hasData && !hasEntries && (
-        <div>
-          <p className="no-data">No entries found</p>
+        <div className="inline-buttons">
+          <p className="no-entries">No entries found</p>
           <button className="button" onClick={resetFilters}>
             <span>Reset filters</span>
           </button>
+          <button
+            className="button"
+            onClick={() =>
+              filterBy({ ...lastFilter, filter: null, value: null })
+            }
+          >
+            <span>Undo last filter</span>
```

Comment:
If there are 0 rows in the dataframe, would we still show this? If so, we need to make sure we actually have `lastFilter` :)

Reply:
I'm not sure I understand your question. We only show this if a filter or a combination of filters results in 0 rows. If the original df has 0 rows, we show "no data" instead.

Reply:
We show "no data" if there are no columns, but it is possible that we have 0 rows and still show columns. Perhaps not with explorer, but I think this could be the case for database queries that return no rows.

Reply:
Gotcha! I'll ensure there are filters!